nativeasset/contract.go

```diff
@@ -105,57 +105,67 @@ func UnpackNativeAssetCallInput(input []byte) (common.Address, common.Hash, *big
 
 // Run implements StatefulPrecompiledContract
 func (c *NativeAssetCall) Run(accessibleState contract.AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
-	if suppliedGas < c.GasCost {
+	env := accessibleState.GetPrecompileEnv()
+	if !env.UseGas(c.GasCost) {
 		return nil, 0, vmerrs.ErrOutOfGas
 	}
-	remainingGas = suppliedGas - c.GasCost
+	ret, err = c.run(env, accessibleState.GetStateDB(), caller, addr, input, readOnly)
+	// This precompile will be wrapped in a libevm `legacy` wrapper, which
```

Comment:
nit do we have a link to this legacy wrapper? Otherwise let's just not use backticks for it (was looking for a variable named `legacy`)

```suggestion
	// This precompile will be wrapped in a libevm legacy wrapper, which
```

Reply:
Unimportant